### PR TITLE
refactor!: new implementation

### LINKED ISSUE #15: Refactor to remove the need for a PassThrough stream
in fastify-static, we need a `PassThrough` object to convert `Send` to an actual `Writable`:

https://github.com/fastify/fastify-static/blob/master/index.js#L113-L140

I think this module should be refactored to in combination with fastify-static to remove the need for the wrapper completely.

### PR DESCRIPTION
**Pull request by climba03003**
Fixes #15 

There is problem I faced in handling `304`, Node.js will filled with un-wanted headers.
So, the test need to remove some assertion on the removal of headers.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by gurgunday — APPROVED** [review]
Thanks for this!

It lgtm overall

**Comment by climba03003** [comment]
I am thinking do we need to provide similar utilities like the current `.pipe`.
It would be as simple as handling the `304` case.

```js
function pipe(send, res) {
  switch(send.statusCode) {
    case 304: {
      // remove Content-* headers
      res.removeHeader('Content-Encoding')
      res.removeHeader('Content-Language')
      res.removeHeader('Content-Length')
      res.removeHeader('Content-Range')
      res.removeHeader('Content-Type')
      break
    }
  }
  res.writeHead(send.statusCode, send.headers)
  send.stream.pipe(res)
}
```

File: lib/send.js [hunk 1]
```diff
@@ -0,0 +1,685 @@
...
+      }
+    }
+  }
+
+  // content-length
+  headers['Content-Length'] = len
+
+  // HEAD support
+  if (request.method === 'HEAD') {
+    return { statusCode, headers, stream: Readable.from('') }
```
Comment (gurgunday) [inline lib/send.js:559]: Not a huge issue, but is there a reason for the empty `Readable` here and not `null`?

Maybe to keep the usage simple (`stream` always pipeable), but it forces  the user to always check the statusCode if they don't want to pipe in [these](https://github.com/fastify/send/blob/bc0be1fa9ff8cf044619b1450c1541917d6bd773/lib/send.js#L442) cases, which is fine, but I think if we just returned `null`, a generic `if` check on `stream` would be faster in higher level libraries like `@fastify/static`

```suggestion
    stream: null
```

Also closer to [Matteo's proposal](https://github.com/fastify/send/issues/15#issuecomment-1386877019):

```js
import http from 'http'
import send from '@fastify/send'
import { pipeline } from 'stream'

const app = http.createServer(async function (req, res) {
  // the options include the duplex options
  const { statusCode, headers, stream } = await send(req, req.url, { root: './' })
  res.writeHead(statusCode, headers)
  if (stream) {
    pipeline(stream, res, function(err) {
      console.log(err)
    })
  } else {
    res.end()
  }
})
```

Comment (climba03003) [reply lib/send.js:559]: I would like to provide a consistent output.
It should always provide `statusCode`, `headers` and `stream`. 
So the users doesn't need to care about when and why the response isn't there.

**Comment by Uzlopak** [comment]
We can now remove http-errors as dependency :)

File: README.md [hunk 2]
```diff
@@ -228,88 +208,9 @@ send.mime.define({
...
 
 var server = http.createServer(function onRequest (req, res) {
-  send(req, parseUrl(req).pathname, { root: '/www/public' })
-    .pipe(res)
-})
-
-server.listen(3000)
-```
-
-### Custom directory index view
```
Comment (mcollina) [inline README.md:238]: is the custom directory index implementation gone?

**Review by mcollina — APPROVED** [review]
lgtm

File: README.md (hunk 2 above)
Comment (climba03003) [reply README.md:238]: It is based on the event API and we might design how to implement it later.

**Review by Uzlopak — CHANGES_REQUESTED** [review]
Remove  http-errors  as dependency ;)

**Review by Uzlopak — APPROVED** [review]
LGTM

**Comment by gurgunday** [comment]
@climba03003 do you have time to refactor `@fastify/static`?

**Pull request merged**